### ISSUE TITLE
Interaction with UI crashes after installing this plugin

### ISSUE THREAD

**Issue by harrythedev** [chromaui/storybook-addon-pseudo-states#3]
Whenever I interact with story with internal state, I'm landing on a crash screen like this.

![image](https://user-images.githubusercontent.com/22530537/106056703-f4ac9000-60bc-11eb-9c94-75b3e2c3b7bc.png)

It seems like this issue is only surfacing when I'm interacting with stories with internal state.
Other stories are working fine.

**Comment by cjevning-chegg** [comment]
Here's a minimal story example to demonstrate the issue:
`
export const button = () => {
  const [count, setCount] = useState(0);
  return (
    <>
      <div>Count: {count}</div>
      <button onClick={() => setCount(count + 1)}>Increment</button>
    </>
  );
};
`
Upon clicking the button, the story crashes to the screen shown above. Confirmed this happens with and without other addons installed, on the latest version of storybook (6.1.15) with the two latest versions of this addon (rc.2 and rc.3)

**Comment by mosherc** [comment]
I am also experiencing this when I click on a button that opens a Modal component in my story. Using React Modal for the component, but not sure that's relevant.

```jsx
export default {
  title: 'Molecules/Modal',
  component: Modal
} as Meta;

const Template: Story<ModalProps> = ({
  children,
  isOpen,
  ...args
}: ModalProps) => {
  const [modal, setModal] = useState(isOpen);
  const toggleModal = (actionStr: string) => (e: any) => {
    action(actionStr)(e);
    setModal(!modal);
  };

  useEffect(() => {
    setModal(isOpen);
  }, [isOpen]);

  // Must do this in consuming app for accessibility features, or pass as prop like below
  Modal.setAppElement('#root');

  return (
    <>
      <p>
        <Button onClick={toggleModal('onClick')}>Open Modal</Button>
      </p>
      <Modal
        isOpen={modal}
        onRequestClose={toggleModal('onRequestClose')}
        // appElement={document.getElementById('root')} this is also valid
        {...args}
      >
        {children}
      </Modal>
    </>
  );
};
```

**Comment by sarahbethfederman** [comment]
If it helps, I'm running into this when I use this when I try to run tests with storybook-addon-performance

**Comment by juliankigwana** [comment]
I get the same even if I remove the Stories that use this. Just adding the plugin alone causes my entire story book to fail with that error.

**Comment by citypaul** [comment]
Unfortunately I'm also seeing this error. When we install the plugin, any interaction with any component appears to break it for the same reason stated above. We're using `@storybook/react` at version `6.4.9`

**Comment by victor-grajski** [comment]
I've been having this issue as well and removing `storybook-addon-pseudo-states` fixed it

**Comment by dmaulick** [comment]
I am also seeing this issue. 

I easily reproduce this crash with any component that uses react state:
```
export const button = () => {
  const [count, setCount] = useState(0);
  return (
    <>
      <div>Count: {count}</div>
      <button onClick={() => setCount(count + 1)}>Increment</button>
    </>
  );
};
```


I did figure out another work around but is not acceptable for my team. You can wrap the stateful component in a dumby wrapper.

And somehow that fixes it...

```
const Button = () => {
  const [count, setCount] = useState(0);
  return (
    <>
      <div>Count: {count}</div>
      <button onClick={() => setCount(count + 1)}>Increment</button>
    </>
  );
};

// This is the actual component being used by the story now
export WrapperButton = () => <Button/>;

```


This is def not idea though

**Comment by dmaulick** [comment]
I wonder if this is the fix: https://github.com/storybookjs/testing-react/issues/30#issuecomment-866336890

**Comment by dmaulick** [comment]
It was because of the `useGlobals` hook.
Upon more research of `useGlobals` it is clear that it is not safe to use everywhere.

Using this technique as a workaround. https://github.com/storybookjs/storybook/issues/12982#issuecomment-865304427

Will post a pull request soon.

**Comment by dmaulick** [comment]
Pull request: https://github.com/chromaui/storybook-addon-pseudo-states/pull/28

**Comment by yannbf** [comment]
Given this was released in [v1.14.0](https://github.com/chromaui/storybook-addon-pseudo-states/releases/tag/v1.14.0) I assume this issue can be closed? cc @ghengeveld

**Comment by ghengeveld** [comment]
Yes 👍
